recipes/bandit/all/conanfile.py

```diff
@@ -1,39 +1,41 @@
 from conan import ConanFile
-from conan.tools.files import apply_conandata_patches, copy, get
+from conan.tools.files import apply_conandata_patches, copy, export_conandata_patches, get
 from conan.tools.layout import basic_layout
 import os
 
-required_conan_version = ">=1.51.3"
+required_conan_version = ">=1.52.0"
 
 
 class BanditConan(ConanFile):
     name = "bandit"
     description = "Human-friendly unit testing for C++11"
-    topics = ("testing", "header-only")
+    license = "MIT"
     url = "https://github.com/conan-io/conan-center-index"
     homepage = "https://github.com/banditcpp/bandit"
-    license = "MIT"
+    topics = ("testing", "header-only")
+    package_type = "header-library"
 
     def export_sources(self):
-        for p in self.conan_data.get("patches", {}).get(self.version, []):
-            copy(self, p["patch_file"], self.recipe_folder, self.export_sources_folder)
+        export_conandata_patches(self)
 
     def requirements(self):
-        self.requires("snowhouse/5.0.0")
+        self.requires("snowhouse/5.0.0", transitive_headers=True)
```

Comment:
As per https://docs.conan.io/2/reference/conanfile/methods/requirements.html#package-type-trait-inferring, this trait is not needed now that the `package_type` is properly defined, but it also does not hurt, and to avoid rebuilding the library for inconsecuentiual change, let's merge it like this unless someone asks for further changes :)